--- amazing0.c	2024-11-22 09:19:02.016271620 +0300
+++ amazing1.c	2024-11-22 09:20:16.112973890 +0300
@@ -4,7 +4,12 @@
 
 #define EMPTY '.'
 #define WALL  '#'
-#define ROOMS_IN_ROW 6
+
+enum ProgramArgs {
+    PROGRAM_NAME = 0,
+    ROOMS_IN_ROW = 1,
+    ARGS_NUM     = 2
+};
 
 struct DSU {
     int *parent;
@@ -192,9 +197,16 @@
 }
 
 int main(int argc, char *argv[]) {
+    if (argc != ARGS_NUM) {
+        printf("Usage: %s <size>\n", argv[PROGRAM_NAME]);
+        exit(1);
+    }
+
     srand(time(NULL));
 
-    struct Maze *maze = Maze__init(EMPTY, WALL, ROOMS_IN_ROW);
+    int rooms_in_row = atoi(argv[ROOMS_IN_ROW]);
+
+    struct Maze *maze = Maze__init(EMPTY, WALL, rooms_in_row);
 
     Maze__build(maze);
 
